src/main/java/uk/gov/companieshouse/authcodenotification/service/PrivateDataRetrievalService.java

```diff
@@ -37,7 +38,7 @@ public OverseasEntityDataApi getOverseasEntityData(String requestId, String comp
         } catch (URIValidationException | IOException e) {
             var message = "Error retrieving overseas entity data from database";
             ApiLogger.errorContext(requestId, message, e, logDataMap.getLogMap());
-            throw new ServiceException(e.getMessage(), e);
+            throw new EntityNotFoundException(e.getMessage(), e);
```

Comment:
If the URI fails validation I don't think it's correct to say the company isn't found (same for an IO exception), so some more thought needed here on the exception handling.

Ideally, we should be reacting a 404 not found code coming back from the API call when throwing your new exception.

Reply:
Followed Mark's advice to catch the exception derived from IOException thrown by the SDK, this can then be checked for the status which is queried to throw the entity not found exception and return 404.

Reply:
Otherwise return the service exception